common/downstream_fatal_error.md

```diff
@@ -0,0 +1,131 @@
+
+## RACF Authorizations
+
+&nbsp;
+
+A common cause of this error is that the user does not have the proper RACF authorizations as outlined in [our dependencies note](../../index). One of the possible causes for this error is the user not having at least `READ` authority to the `IRR.IRRSMO00.PRECHECK` resource in the `XFACILIT` class, which is required for `set` or `alter` operations. See the [Check for & set up RACF Authorizations](../setup_precheck) documentation for more details. Another possible cause of this error is the user not having `ALTER` authority to the `userid.IRRSMO00` resource in the `SURROGAT` class, which is required to make security requests as another userid. Please review our [run as userid](../run_as_userid) documentation for more information on this feature.
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin()
+
+try:
+    user_admin.set_uid("squidwrd", 123456)
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output
+```console
+Security request made to IRRSMO00 failed.
+
+SAF Return Code: 8 
+RACF Return Code: 200 
+RACF Reason Code: 16
+
+Check to see if the proper RACF permissions are in place.
+For `set` or `alter` functions, you must have at least READ access to `IRR.IRRSMO00.PRECHECK` in the `XFACILIT` class.
+```
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin(run_as_userid="ESWIFT")
+
+try:
+    user_admin.set_uid("squidwrd", 123456)
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output
+```console
+Security request made to IRRSMO00 failed.
+
+SAF Return Code: 8 
+RACF Return Code: 200 
+RACF Reason Code: 8
+
+Check to see if the proper RACF permissions are in place.
+For the `run_as_userid` feature, you must have at least UPDATE access to `ESWIFT.IRRSMO00` in the `SURROGAT` class.
+```
+
+## Improper Use of Traits
+
+&nbsp;
+
+As stated, `DownstreamFatalError` may also be raised if there is a problem with the data pyRACF passes to IRRSMO00. In this situation, you can review `DownstreamFatalError.request_xml` and the `DownwstreamFatalError.result_dictionary` for more information.
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin()
+
+try:
+    user_admin.alter("squidwrd", traits={"base:special": "STRING"})
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output
+```console
+Security request made to IRRSMO00 failed.
+
+SAF Return Code: 8 
+RACF Return Code: 2000 
+RACF Reason Code: 76
+
+See results dictionary 'DownstreamFatalError.result' for more details.

```

Comment:
I think this message is actually wrong. I think it should be `DownstreamFatalError.result_dictionary`. This update will require an update to the code as well. But you can probably just open a very small pull request with those updates since we already merged.

Reply:
Problem resolved through documentation changes. No code changes are required since result dictionary is actually stored in the `DownstreamFatalError.result` class attribute and not the `DownstreamFataError.result_dictionary` class attribute.